### PR TITLE
Create Feature less module in UI code and import alias

### LINKED ISSUE #758: [UI] Support less module in UI code and import alias
Currently all styles are not well managed in UI code, should enable less module and alias import to make style manageable and reusable.

### PR DESCRIPTION
**Pull request by Fendoe**
## Description

Resolves #758

**Add dependencies：**

1. @ant-design/icons
2. classnames

**Add devDependencies：**

1. @craco/craco
2. babel-plugin-import
3. craco-less
4. eslint-import-resolver-typescript 
5. eslint-plugin-import
6. eslint-plugin-prettier

**Modify the scripts field in package.json.**
```
"scripts": {
-   "start": "react-scripts start",
-   "build": "react-scripts build",
-   "test": "react-scripts test",
+   "start": "craco start",
+   "build": "craco build",
+   "test": "craco test",
}
```

**Add craco.config.js**

``` js
const path = require("path");

const { loaderByName } = require("@craco/craco");
const CracoLessPlugin = require("craco-less");

const resolve = (dir) => path.resolve(__dirname, dir);

module.exports = {
  babel: {
    plugins: [
      [
       // Import Antd components on demand
        "import", 
        {
          libraryName: "antd",
          libraryDirectory: "es",
          style: true,
        },
      ],
    ],
  },
  webpack: {
    alias: {
      "@": resolve("src"),
    },
  },
  plugins: [
    {
     // Support less, less module 
      plugin: CracoLessPlugin,
      options: {
        lessLoaderOptions: {
          lessOptions: {
            modifyVars: {},
            javascriptEnabled: true,
          },
        },
        modifyLessModuleRule(lessModuleRule, context) {
          // Configure the file suffix
          lessModuleRule.test = /\.module\.less$/;

          // Configure the generated local ident name.
          const cssLoader = lessModuleRule.use.find(loaderByName("css-loader"));
          cssLoader.options.modules = {
            localIdentName: "[local]_[hash:base64:5]",
          };

          return lessModuleRule;
        },
      },
    },
  ],
};

```

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to clarify your proposed changes.

### CODE REVIEW

File: ui/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+[{.eslintrc,.babelrc,.stylelintrc,*.json}]
+indent_style=space
+indent_size=2
+
+[*.less]
+indent_style=space
+indent_size=2
+
+[*.{js,jsx,ts,tsx,vue}]
```
Comment (blrchen) [inline ui/.editorconfig:19]: vue can be removed to avoid confusion

File: ui/.editorconfig [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+root = true
+
+[*]
+charset=utf-8
+end_of_line=lf
+insert_final_newline=false
+indent_style=space
+indent_size=2
+
+[{.eslintrc,.babelrc,.stylelintrc,*.json}]
```
Comment (blrchen) [inline ui/.editorconfig:11]: Line 11 to 21 can be removed as already defined in *

File: ui/.eslintrc [hunk 3]
```diff
@@ -4,23 +4,67 @@
...
+    //       }
+    //     ],
+    //     "pathGroupsExcludedImportTypes": ["react"],
+    //     "newlines-between": "always",
+    //     "alphabetize": {
+    //       "order": "asc",
+    //       "caseInsensitive": true
+    //     }
+    //   }
+    // ]
```
Comment (blrchen) [inline ui/.eslintrc:67]: Please delete instead of comment

File: ui/src/pages/home/index.module.less [hunk 4]
```diff
@@ -0,0 +1,28 @@
+.home {
+  :global {
+    .ant-card {
+      box-shadow: 5px 8px 15px 5px rgba(208, 216, 243, 0.6);
+      border-radius: 8px;
+    }
+  }
+
+  .cardMeta {
```
Comment (blrchen) [inline ui/src/pages/home/index.module.less:9]: Please use class-name pattern, which is the standard css naming convention.

**Comment by blrchen** [comment]
Thanks @Fendoe for contribution, please add some docs for craco and less

**Comment by Fendoe** [comment]
> For @antdesign-icons - is this required? Or we can change to a different icon lib?

We use @antdesign-icons all the time. It just doesn't add dependencies.
We can cheange it in the futher.

File: ui/package.json [hunk 5]
```diff
@@ -15,18 +17,24 @@
     "react-router-dom": "^6.3.0"
   },
   "devDependencies": {
+    "@craco/craco": "^7.0.0-alpha.8",
```
[the remark this answers is not in the document]
Comment (Fendoe) [reply ui/package.json:20]: It is already the latest version.

File: ui/src/pages/home/index.module.less (hunk 4 above)
Comment (Fendoe) [reply ui/src/pages/home/index.module.less:9]: The less module needs to interact with tsx, so camelCase is used to facilitate tsx invocation.

**Review by blrchen — APPROVED** [review]

**Review by Yuqing-cat — APPROVED** [review]

**Review by blrchen — APPROVED** [review]

**Pull request merged**